File: backend/src/main/java/com/sajjad/academicportalbackend/model/config/StudentProfile.java
package com.sajjad.academicportalbackend.model.config;


import com.sajjad.academicportalbackend.model.auth.User;
import com.sajjad.academicportalbackend.model.super_classes.AuditableEntity;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Entity
@NoArgsConstructor
public class StudentProfile extends AuditableEntity {
    @Getter
    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id")
    private User user;
    private String roll;
    private String firstName;
    private String lastName;
    private String contactNumber;
    private String department;
    private String session;

}

File: frontend/src/app/model/config/student-profile.ts
import { User } from "../auth/user";
import { AuditAbleModel } from "../super-model/audit-able-model";

export interface StudentProfile extends AuditAbleModel {
    // id?: number;
    user?: User;
    firstName?: string;
    lastName?: string;
    contactNumber?: string;
    department?: string;
    roll?: string
    session?: string;
    
   

    // Assuming you might want to represent the related user in some way in the frontend:
    // userId?: number; // Or another User interface/type if you have it.
}
